src/test/java/BaseTest.java

```diff
@@ -26,16 +32,22 @@ public static void chromeConfigs() {
 
     @BeforeClass
     @Parameters({"BaseURL"})
-    public void launchBrowser(String BaseURL) {
-        ChromeOptions chromeOptions = new ChromeOptions();
-        chromeOptions.addArguments("--disable-notifications");
-
-        driver = new ChromeDriver(chromeOptions);
+    public void launchBrowser(String BaseURL) throws MalformedURLException {
+       // ChromeOptions chromeOptions = new ChromeOptions();
+       // chromeOptions.addArguments("--disable-notifications");
+//System.setProperty("webdriver.gecko.driver", "geckodriver");
+//driver= new FirefoxDriver();
+       driver = pickBrowser(System.getProperty("browser"));
```

Comment:
**Step 1** Add the geckodriver.exe file to the project
Download GeckoDriver  https://github.com/mozilla/geckodriver/releases
![image](https://user-images.githubusercontent.com/104761482/201294016-ad1b1ba5-47dc-451b-9e6c-b59ea719cf5d.png)

![image](https://user-images.githubusercontent.com/104761482/201294078-cbe8148b-c077-4c95-abc6-239ee6e5e435.png)

**Step 2**  Add "geckodriver.exe"
For example
`System.setProperty("webdriver.gecko.driver","geckodriver.exe");`
![image](https://user-images.githubusercontent.com/104761482/201294239-cb6ccd25-6531-405c-ac9b-bb2fc09cbdda.png)


**Step 3**  Write a method to initialize the driver, for example
`driver = pickBrowser("firefox");`
![image](https://user-images.githubusercontent.com/104761482/201294328-fab7a7af-3f50-49ea-88fb-e990d206cce6.png)

**1. To start the selenium server:**

For example java `-jar selenium-server-4.6.0.jar standalone`
![image](https://user-images.githubusercontent.com/104761482/201294494-6de41193-36c9-4df5-8dd1-240fe3138989.png)

2. Add an URL to your project, for example

![image](https://user-images.githubusercontent.com/104761482/201294734-4c663d2b-0dbc-42d7-98fe-2b8975d3b66a.png)

`String gridURL = "http://172.21.160.1:4444/";`

![image](https://user-images.githubusercontent.com/104761482/201294845-e5f2b224-e97b-4dae-b98c-fd7928cd4e2e.png)

![image](https://user-images.githubusercontent.com/104761482/201294920-0f93d1b0-7c18-4c04-bd5d-b8193cb8ebbb.png)

________________________________________________________________________________________________________________________

```
public class BaseTest {

  protected WebDriver driver;
  protected String myEmail = "holostenco.yuliya@gmail.com";
  protected String myPassword = "te$t$tudent";
  String url;


  @BeforeSuite
  public static void chromeConfigs() {
    if (System.getProperty("os.name").toLowerCase().contains("win")) {
      System.setProperty("webdriver.chrome.driver", "chromedriver.exe");
    }
  }

  @BeforeClass
  @Parameters({"BaseURL"})
  public void launchBrowser(String BaseURL) throws MalformedURLException {
    driver = pickBrowser("firefox");
    driver.manage().window().maximize();
    driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
    url = BaseURL;
    driver.get(url);
  }


  public WebDriver pickBrowser(String browser) throws MalformedURLException {
    DesiredCapabilities capabilities = new DesiredCapabilities();
    String gridURL = "http://172.21.160.1:4444/";
    switch (browser) {
      case "firefox":
        System.setProperty("webdriver.gecko.driver", "geckodriver.exe");
        return driver = new FirefoxDriver();
      case "safari":
        return driver = new SafariDriver();
      case "grid-safari":
        capabilities.setCapability("browserName", "safari");
        return driver = new RemoteWebDriver(URI.create(gridURL).toURL(), capabilities);
      case "grid-firefox":
        capabilities.setCapability("browserName", "firefox");
        return driver = new RemoteWebDriver(URI.create(gridURL).toURL(), capabilities);
      case "grid-chrome":
        capabilities.setCapability("browserName", "chrome");
        return driver = new RemoteWebDriver(URI.create(gridURL).toURL(), capabilities);
      default:
        return driver = new ChromeDriver();

    }
  }

  @AfterMethod
  public void tearDownBrowser() {
    driver.quit();
  }
}
```